go/client/cmd_ctl_watchdog2.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package client
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/keybase/cli"
+	"github.com/keybase/client/go/install"
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/go-updater/watchdog"
+)
+
+// CmdWatchdog2 defines watchdog command
+type CmdWatchdog2 struct {
+	libkb.Contextified
+}
+
+// ParseArgv is args for the watchdog command
+func (c *CmdWatchdog2) ParseArgv(ctx *cli.Context) error {
+	return nil
+}
+
+// Run watchdog
+func (c *CmdWatchdog2) Run() error {
+	env, log := c.G().Env, c.G().Log
+	log.Info("Starting watchdog")
+	runMode := env.GetRunMode()
+	if runMode != libkb.ProductionRunMode {
+		return fmt.Errorf("Watchdog is only supported in production")
+	}
+
+	// Service
+	keybasePath, err := install.BinPath()
+	if err != nil {
+		return err
+	}
+	serviceLogPath := filepath.Join(env.GetLogDir(), libkb.ServiceLogFileName)
+	serviceProgram := watchdog.Program{
+		Path: keybasePath,
+		Args: []string{
+			"-d",
+			"--log-file=" + serviceLogPath,
+			"service",
+			"--watchdog-forked",
+		},
+		ExitOn: watchdog.ExitOnSuccess,
+	}
+
+	// KBFS
+	kbfsPath, err := install.KBFSBinPath(runMode, "")
+	if err != nil {
+		return err
+	}
+	mountDir, err := env.GetMountDir()
+	if err != nil {
+		return err
+	}
+	kbfsProgram := watchdog.Program{
+		Path: kbfsPath,
+		Args: []string{
+			"-debug",
+			"-log-to-file",
+			mountDir,
+		},
+	}
+
+	// Updater
+	updaterPath, err := install.UpdaterBinPath()
+	if err != nil {
+		return err
+	}
+	updaterProgram := watchdog.Program{
+		Path: updaterPath,
+		Args: []string{
+			"-path-to-keybase=" + keybasePath,
+		},
+	}
+
+	// Start and monitor all the programs
+	programs := []watchdog.Program{
+		serviceProgram,
+		kbfsProgram,
+		updaterProgram,
+	}
+	if err := watchdog.Watch(programs, 10*time.Second, c); err != nil {
+		return err
+	}
+
+	for {
+		time.Sleep(time.Hour)
```

Comment:
This is a wait forever since `watchdog.Watch` is non-blocking... we can change to `select{}` which also does a wait forever in go without sleep